src/Nethermind/Nethermind.Blockchain/Producers/TxPoolTxSource.cs

```diff
@@ -37,18 +38,22 @@ public class TxPoolTxSource : ITxSource
     {
         private readonly ITxPool _transactionPool;
         private readonly IStateReader _stateReader;
+        private readonly ITransactionComparerProvider _transactionComparerProvider;
         private readonly ITxFilter _txFilter;
+        private readonly ISpecProvider _specProvider;
         protected readonly ILogger _logger;
 
-        public TxPoolTxSource(ITxPool transactionPool, IStateReader stateReader, ILogManager logManager, ITxFilter txFilter = null)
+        public TxPoolTxSource(ITxPool transactionPool, IStateReader stateReader, ISpecProvider specProvider, ITransactionComparerProvider transactionComparerProvider, ILogManager logManager, ITxFilter txFilter = null)
         {
             _transactionPool = transactionPool ?? throw new ArgumentNullException(nameof(transactionPool));
             _stateReader = stateReader ?? throw new ArgumentNullException(nameof(stateReader));
-            _txFilter = txFilter ?? new MinGasPriceTxFilter(UInt256.Zero);
+            _transactionComparerProvider = transactionComparerProvider ?? throw new ArgumentNullException(nameof(transactionComparerProvider));
+            _txFilter = txFilter ?? new MinGasPriceTxFilter(UInt256.Zero, specProvider);
+            _specProvider = specProvider ?? throw new ArgumentNullException(nameof(specProvider));
             _logger = logManager?.GetClassLogger<TxPoolTxSource>() ?? throw new ArgumentNullException(nameof(logManager));
         }
         
-        public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimit)
+        public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimit, UInt256 baseFee)
```

Comment:
why basefee is not handled by filters here?

Reply:
I feel like we start leaking interfaces into multiple levels. Who has responsibility of filtering transactions? Why TxPoolTxSource? This is just a source right? Is it even meant to filter?